src/boot/sysfail.c

```diff
@@ -3,10 +3,37 @@
 #include "sysfail.h"
 #include "util.h"
 
-SysFailType sysfail_check(void) {
+
+static bool check_firmware_update_failure(void) {
```

Comment:
boolean function named `check_xyz_failure()` sounds confusing in the result. Without reading the function definition, we cannot know the meaning of `true/false`.
Please rename e.g. `find_firmware_update_failure()` or `firmware_update_is_failed()`.